packages/quantic/force-app/main/default/lwc/quanticGeneratedAnswer/quanticGeneratedAnswer.js

```diff
@@ -297,16 +315,16 @@ export default class QuanticGeneratedAnswer extends LightningElement {
    */
   async handleDislike(event) {
     event.stopPropagation();
-    this.generatedAnswer.dislike?.();
-    if (!this.feedbackSubmitted) {
+    if (!this.feedbackSubmitted && !this._disliked) {
+      this._disliked = true;
+      this._liked = false;
+      this.generatedAnswer.dislike?.();
       // @ts-ignore
-      await FeedbackModal.open({
-        label: this.labels.feedback,
+      await FeedbackModalQna.open({
         size: 'small',
-        description: this.labels.feedback,
-        options: this.options,
+        liked: false,
+        label: this.labels.feedbackHelpUsImprove,
         handleSubmit: this.submitFeedback.bind(this),
-        optionsLabel: this.labels.whyGeneratedAnswerWasNotHelpful,
       });
```

Comment:
We could isolate the logic to open the modal in a separate function so we can just call it for the like handler and the dislike handler instead of having to write this same logic twice.

Reply:
I don't see much value in doing this, especially when we have at least 1 difference